Add Prompting 101 Tile to independent-ai-topics (#1)
Ada-Developers-Academy/independent-ai-topics: 02-prompting-101/improving-chatGPTs-responses.md

We can check out an example – while they are improving, Generative AI tools commonly struggle with math concepts. In ChatGPT, try a prompt like:

> Is 6113 a prime number?

We are likely to see an answer similar to:

> To determine if 6113 is a prime number, we need to check if it has any divisors other than 1 and itself. 
> 
> Let's find its divisors:
>
> 1. 6113 ÷ 1 = 6113
> 2. 6113 ÷ 7 = 873
> 
> So, 6113 is divisible by 1 and 7, in addition to itself and 1. Since it has divisors other than 1 and itself, 6113 is not a prime number.

But this is incorrect, 6,113 *is* a prime number!

### !callout-info

## Try other prime numbers if the prompt works on the first try
If ChatGPT gets it right on the first time, you can change to a different prime number until you get an incorrect result. Because models are receiving feedback from users and chats are also used to further train models, commonly asked questions that reliably give us wrong results at one point may change over time. You can do a web search on "lists of prime numbers" or "prime number calculator" to get started quickly.

### !end-callout

> This incorrect answer is clearly not desired, but we can often get to a correct response for this question by asking the AI tool to show it's work and reason itself to a correct answer. Try out this updated prompt in either the same chat or a new one:

Reviewer: ```suggestion
This incorrect answer is clearly not desired, but we can often get to a correct response for this question by asking the AI tool to show its work and reason itself to a correct answer. Try out this updated prompt in either the same chat or a new one:
```

Much better! The topics are correct, and we have some helpful hashtags. It turns out that the message doesn't meet all our needs though, it's 257 characters long, which breaks our request that the message be 250 characters or less. We could try regenerating the response or adjusting our wording to get ChatGPT to respect all our constraints.

</details>  

## Giving Examples/Shot Prompting

As we’ve learned, generative AIs act very much like predictive text, but with a lot more information to use to make their predictions. No matter how much data they have though, there will always be specific tasks or response formats they they will not have encountered. 

Especially when you need the output formatted a particular way or in a way that is difficult to describe, we can improve an AI Tool’s response by including examples (called “Shots” when talking about prompt engineering) of what it is that we want the AI tool to do. When we include a single example, this is called “Single Shot” prompting, and when we give it 2 or more examples it gets called “Few Shot” prompting.

> Let's say that we moderate a forum and want to categorize feedback from our users. We're going start from a zero-shot prompt and iterate on it until we have a few-shot prompt. For our example, we have a list of feedback like:

Author: ```suggestion
We're going to look at a prompt that's different from others we've seen – this time we're not going to include any instructions telling the generative AI what it should do. Let's say that we moderate a forum and want to categorize feedback from our users. We're going start from a zero-shot prompt and iterate on it until we have a few-shot prompt. For our example, we have a list of feedback like:
```

> Feedback 1: I love this site but I can’t log in on iPhone!  
> Feedback 2: The custom icons are so cute.  
> Feedback 3: The cursor disappears after clicking the logo.  

and as a response we want to get back:

> Feedback 1: "I love this site but I can’t log in on iPhone!"  
> Sentiment: negative  
> Themes: iPhone login  
> 
> Feedback 2: "The custom icons are so cute."  
> Sentiment: positive  
> Themes: custom icons  
> 
> Feedback 3: "The cursor disappears after clicking the logo."  
> Sentiment: negative  
> Themes: UI, cursor  

> If we include no examples, no instructions, and simply submit our list of feedback as a prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 

Reviewer: I was a bit lost reading this section because I wasn't entirely sure what the prompt to Chat was - is it literally the list of feedback with no other context?

Author: That's exactly what's happening, the prompt is literally just the feedback. Would it be helpful to include a sentence before we show the example around "a prompt can be just data without instructions"?

Author: Edit: I added the line as a suggestion on the PR so it's easier to see/review.

I'm thinking above the feedback when introducing the example, we could start the paragraph with something like:

> We're going to look at a prompt that's different from others we've seen – this time we're not going to include _any_ direct instructions on what the generative AI should do. Let's say that we moderate a forum...

> If we include no examples, no instructions, and simply submit our list of feedback as a prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 

Author: ```suggestion
If we include no examples, no instructions, and submit our list of feedback itself as the prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 
```
